language-support/scala/bindings/src/main/scala/com/digitalasset/ledger/client/binding/binding/Primitive.scala

```diff
@@ -187,26 +163,19 @@ private[client] object OnlyPrimitive extends Primitive {
     override def apply[Tpl <: Template[Tpl]](
         packageId: String,
         moduleName: String,
-        entityName: String): TemplateId[Tpl] =
+        entityName: String
+    ): TemplateId[Tpl] =
       ApiTypes.TemplateId(
-        rpcvalue.Identifier(
-          packageId = packageId,
-          name = s"$moduleName.$entityName",
-          moduleName = moduleName,
-          entityName = entityName))
+        rpcvalue
+          .Identifier(packageId = packageId, moduleName = moduleName, entityName = entityName))
 
     private[binding] override def substEx[F[_]](fa: F[rpcvalue.Identifier]) =
       ApiTypes.TemplateId subst fa
 
-    override def unapply[Tpl](t: TemplateId[Tpl]): Option[(String, String, String)] =
-      // TODO SC DEL-6727 use this instead with daml-lf value interface
-      // rpcvalue.Identifier unapply t.unwrap
-      t.unwrap match {
-        case rpcvalue.Identifier(packageId, _, moduleName, entityName)
-            if moduleName.nonEmpty && entityName.nonEmpty =>
-          Some((packageId, moduleName, entityName))
-        case _ => None
-      }
+    override def unapply[Tpl](t: TemplateId[Tpl]): Some[(String, String, String)] = {
+      val rpcvalue.Identifier(packageId, moduleName, entityName) = t.unwrap
```

Comment:
No we don't check anymore that `moduleName` and `entityName` are non-empty.

Reply:
For what I understand, the test was used to check if it is a new/old identifier.

Reply:
Okay, but a templateId is only valid if the individual strings coming from the wire aren't empty. But maybe this is not the place to check this.

Reply:
The check is done inside the ledger API.
We probably do not need to duplicate the test in Scala binding side.